src/Compilers/VisualBasic/Portable/Symbols/PropertySymbol.vb

```diff
@@ -175,28 +175,34 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Symbols
                     Return False
                 End If
 
-                Select Case receiverOpt.Kind
-                    Case BoundKind.WithLValueExpressionPlaceholder
-                        ' This can be a `Me` reference used as a target for a `With` statement
-                        Dim currentBinder As Binder = containingBinder
-
-                        While currentBinder IsNot Nothing AndAlso currentBinder.ContainingMember Is containingMember
-                            Dim withBlockBinder = TryCast(currentBinder, WithBlockBinder)
-                            If withBlockBinder IsNot Nothing Then
-                                Return withBlockBinder.Info?.ExpressionPlaceholder Is receiverOpt AndAlso
-                                       withBlockBinder.Info.OriginalExpression.Kind = BoundKind.MeReference
+                If receiverOpt.Kind = BoundKind.WithLValueExpressionPlaceholder OrElse receiverOpt.Kind = BoundKind.WithRValueExpressionPlaceholder Then
+                    ' This can be a reference used as a target for a `With` statement
+                    Dim currentBinder As Binder = containingBinder
+
+                    While currentBinder IsNot Nothing AndAlso currentBinder.ContainingMember Is containingMember
+                        Dim withBlockBinder = TryCast(currentBinder, WithBlockBinder)
+                        If withBlockBinder IsNot Nothing Then
+                            If withBlockBinder.Info?.ExpressionPlaceholder Is receiverOpt Then
+                                receiverOpt = withBlockBinder.Info.OriginalExpression
                             End If
 
-                            currentBinder = currentBinder.ContainingBinder
-                        End While
+                            Exit While
+                        End If
 
-                        Return False
+                        currentBinder = currentBinder.ContainingBinder
+                    End While
+                End If
 
-                    Case BoundKind.MeReference, BoundKind.MyBaseReference, BoundKind.MyClassReference
-                        Return True
-                    Case Else
-                        Return False
-                End Select
+                Do
+                    Select Case receiverOpt.Kind
+                        Case BoundKind.MeReference, BoundKind.MyBaseReference, BoundKind.MyClassReference
+                            Return True
+                        Case BoundKind.Parenthesized
```

Comment:
Looks like this fixes the handling for parenthesized receiver in `With`. Do we have a test for that case?

Reply:
>Looks like this fixes the handling for parenthesized receiver in With. Do we have a test for that case?

EvaluationInitOnlySetter_10

---
In reply to: [566277477](https://github.com/dotnet/roslyn/pull/50770#discussion_r566277477) [](ancestors = 566277477)